package/ios/Core/Types/CodeScannerOptions.swift

```diff
@@ -16,8 +16,18 @@ struct CodeScannerOptions: Equatable {
 
   init(fromJsValue dictionary: NSDictionary) throws {
     if let codeTypes = dictionary["codeTypes"] as? [String] {
-      self.codeTypes = try codeTypes.map { value in
-        return try AVMetadataObject.ObjectType(withString: value)
+      self.codeTypes = try codeTypes.compactMap { value in
+        do {
+          return try AVMetadataObject.ObjectType(withString: value)
+        } catch let CameraError.codeScanner(codeScannerError) {
+          switch codeScannerError {
+          case .iosVersionNotSupported:
+            VisionLogger.log(level: .warning, message: codeScannerError.message)
```

Comment:
There's no point in throwing an error if we just ignore it later.

Please add the `ios-version-not-supported` error to the TypeScript error types, and then in here we just let the error propagate upwards for the user to handle.

Reply:
but then hmm he will just ignore it later anyways as well

Reply:
Is it okay to propagate the error to the JS side? In that case, the `onError` callback will fire, and the user can handle the error. However, the user might assume that this callback fired because the camera failed to start and cannot operate, leading them to show an error screen, for instance. But the scanner can still operate.

I can refactor the solution without throwing an error

Reply:
@mrousavy 

I've refactored without throwing an error. Also i've added supported code types table to the docs